File: java/src/main/java/com/sospos/model/ItemCarrito.java
package com.sospos.model;

public class ItemCarrito {
    private Producto producto;
    private int cantidad;

    public ItemCarrito(Producto producto) {
        this.producto = producto;
        this.cantidad = 1;
    }

    public Producto getProducto() { return producto; }
    public int getCantidad() { return cantidad; }
    public void setCantidad(int cantidad) { this.cantidad = cantidad; }
    public double getSubtotal() { return producto.getPrecio() * cantidad; }
}

File: dotnet/SosPOS/Models/ItemCarrito.cs
namespace SosPOS.Models;

public class ItemCarrito
{
    public Producto Producto { get; set; } = null!;
    public string Nombre => Producto.Nombre;
    public int Cantidad { get; set; }
    public decimal Total => Producto.Precio * Cantidad;
    public string TotalFormateado => $"${Total:F2}";
}
